cid.go

```diff
@@ -601,3 +581,42 @@ func PrefixFromBytes(buf []byte) (Prefix, error) {
 		MhLength: int(mhlen),
 	}, nil
 }
+
+func CidFromBytes(data []byte) (int, Cid, error) {
+	if len(data) > 2 && data[0] == 18 && data[1] == 32 {
+		if len(data) < 34 {
+			return 0, Undef, fmt.Errorf("not enough bytes for cid v0")
+		}
+
+		h, err := mh.Cast(data[:34])
+		if err != nil {
+			return 0, Undef, err
+		}
+
+		return 34, NewCidV0(h), nil
+	}
+
+	vers, n := binary.Uvarint(data)
+	if err := uvError(n); err != nil {
+		return 0, Undef, err
+	}
+
+	if vers != 1 {
+		return 0, Undef, fmt.Errorf("expected 1 as the cid version number, got: %d", vers)
+	}
+
+	_, cn := binary.Uvarint(data[n:])
+	if err := uvError(cn); err != nil {
+		return 0, Undef, err
+	}
+
+	mhr := mh.NewReader(bytes.NewReader(data[n+cn:]))
+	h, err := mhr.ReadMultihash()
```

Comment:
Is this any better than `mh.Cast(data[n+cn:])`?

Reply:
mh.Cast didnt work at all with trailing bytes, the Reader did, but allocated a lot more. I implemented a method in multihash to do just what we want here, and am using it now.